master/internal/usergroup/api_groups.go

```diff
@@ -0,0 +1,216 @@
+package usergroup
+
+import (
+	"context"
+
+	"github.com/uptrace/bun"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/groupv1"
+)
+
+type ApiServer struct{}
+
+func (a *ApiServer) CreateGroup(ctx context.Context, req *apiv1.CreateGroupRequest,
+) (resp *apiv1.GroupWriteResponse, err error) {
+	// Detect whether we're returning special errors and convert to gRPC error
+	defer func() {
+		err = mapAndFilterErrors(err)
+	}()
+
+	group := Group{
+		Name: req.Name,
+	}
+
+	createdGroup, err := AddGroup(ctx, group)
+	if err != nil {
+		return nil, err
+	}
+
+	uids := intsToUserIDs(req.AddUsers)
+	err = AddUsersToGroup(ctx, createdGroup.ID, uids...)
```

Comment:
Good catch on the atomicity. Fixed!

I'm not quite understanding what the particular problem is with the error messages. Does the CLI merely print the exact error message returned by the API or something?

Reply:
The CLI for the most part prints the exact message returned by the API. 

However regardless the API is exposed to end users and personally if I am consuming an API would much rather see something like ```could not create group: failed to add users with ID 9998,9999 not found```. 

If I had never used Determined the error message currently returned would make me wonder if I even have the right endpoint

Reply:
@NicholasBlaskey Great points and we changed some of the error handling to allow us to make more descriptive error messages. There is a technical limitation on returning more descriptive information on bulk operations because we won't know for sure which specific item failed without adding some more expensive code, so we added a more generic message for those operations (batch add/remove users). Hopefully we get good mileage out of what we have now though.